### DISCUSSION TITLE
error when combining two matrices

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#1375] [General]
Hi @rcorces 

When combining the GeneScoreMatrix with the Motif Matrix using the script
trajCombined <- trajGSM2
assay(trajCombined) <- t(apply(assay(trajGSM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

I encounter the error that reads as follows:
Error in `assays<-`(`*tmp*`, withDimnames = withDimnames, ..., value = `*vtmp*`) : 
  please use 'assay(x, withDimnames=FALSE)) <- value' or 'assays(x, withDimnames=FALSE))
  <- value' when the rownames or colnames of the supplied assay(s) are not identical to
  those of the receiving SummarizedExperiment object 'x'

Can you suggest a way to modify the script to prevent this error?

The trajectory heatmap of the combined matrices are plotted with no problem but the recurrence of this error is bothersome.

Thanks!

**Comment by rcorces** [comment]
have you tried doing what the error suggests?
```
assay(trajCombined, withDimnames = FALSE) <- t(apply(assay(trajGSM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))
```

**Reply by sylestiel** [reply]
Okay. That worked. Why was that necessary?

**Reply by rcorces** [reply]
as the error says, this is required when:

> when the rownames or colnames of the supplied assay(s) are not identical to those of the receiving SummarizedExperiment object 'x'

**Reply by sylestiel** [reply]
Thanks!
